File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/RecordCreateForms/ResolutionEditBox.vb
Imports System
Imports System.Windows.Forms

Namespace RecordCreateForms

	Partial Public Class ResolutionEditBox
		Inherits UserControl
#Region "Private fields"

		Private _currentUnit As ResolutionUnits.Unit

#End Region

#Region "Public constructor"

		Public Sub New()
			InitializeComponent()

			cbScaleUnits.Items.AddRange(ResolutionUnits.Units.ToArray())

			RawValue = 0
			_currentUnit = ResolutionUnits.PpmUnit
			cbScaleUnits.SelectedItem = _currentUnit
		End Sub

#End Region

#Region "Public properties"

		Public Property RawValue() As Double
			Get
				Return Double.Parse(tbValue.Text)
			End Get
			Friend Set(ByVal value As Double)
				tbValue.Text = value.ToString()
			End Set
		End Property

		Public Property PpmValue() As Double
			Get
				Return ResolutionUnits.Unit.Convert(_currentUnit, ResolutionUnits.PpmUnit, RawValue)
			End Get
			Set(ByVal value As Double)
				cbScaleUnits.SelectedItem = ResolutionUnits.PpmUnit
				RawValue = value
			End Set
		End Property

		Public Property PpcmValue() As Double
			Get
				Return ResolutionUnits.Unit.Convert(_currentUnit, ResolutionUnits.PpcmUnit, RawValue)
			End Get
			Set(ByVal value As Double)
				cbScaleUnits.SelectedItem = ResolutionUnits.PpcmUnit
				RawValue = value
			End Set
		End Property

		Public Property PpmmValue() As Double
			Get
				Return ResolutionUnits.Unit.Convert(_currentUnit, ResolutionUnits.PpmmUnit, RawValue)
			End Get
			Set(ByVal value As Double)
				cbScaleUnits.SelectedItem = ResolutionUnits.PpmmUnit
				RawValue = value
			End Set
		End Property

		Public Property PpiValue() As Double
			Get
				Return ResolutionUnits.Unit.Convert(_currentUnit, ResolutionUnits.PpiUnit, RawValue)
			End Get
			Set(ByVal value As Double)
				cbScaleUnits.SelectedItem = ResolutionUnits.PpiUnit
				RawValue = value
			End Set
		End Property

#End Region

#Region "Private form events"

		Private Sub CbScaleUnitsSelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbScaleUnits.SelectedIndexChanged
			If cbScaleUnits.SelectedItem IsNot _currentUnit Then
				Dim newValue As Double = ResolutionUnits.Unit.Convert(_currentUnit, CType(cbScaleUnits.SelectedItem, ResolutionUnits.Unit), RawValue)
				RawValue = newValue
				_currentUnit = CType(cbScaleUnits.SelectedItem, ResolutionUnits.Unit)
			End If
		End Sub

#End Region
	End Class
End Namespace

File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/RecordCreateForms/ResolutionEditBox.cs
using System;
using System.Windows.Forms;

namespace Neurotec.Samples.RecordCreateForms
{
	public partial class ResolutionEditBox : UserControl
	{
		#region Private fields

		ResolutionUnits.Unit _currentUnit;

		#endregion

		#region Public constructor

		public ResolutionEditBox()
		{
			InitializeComponent();

			cbScaleUnits.Items.AddRange(ResolutionUnits.Units.ToArray());

			RawValue = 0;
			cbScaleUnits.SelectedItem = _currentUnit = ResolutionUnits.PpmUnit;
		}

		#endregion

		#region Public properties

		public double RawValue
		{
			get
			{
				return double.Parse(tbValue.Text);
			}
			internal set
			{
				tbValue.Text = value.ToString();
			}
		}

		public double PpmValue
		{
			get
			{
				return ResolutionUnits.Unit.Convert(_currentUnit, ResolutionUnits.PpmUnit, RawValue);
			}
			set
			{
				cbScaleUnits.SelectedItem = ResolutionUnits.PpmUnit;
				RawValue = value;
			}
		}

		public double PpcmValue
		{
			get
			{
				return ResolutionUnits.Unit.Convert(_currentUnit, ResolutionUnits.PpcmUnit, RawValue);
			}
			set
			{
				cbScaleUnits.SelectedItem = ResolutionUnits.PpcmUnit;
				RawValue = value;
			}
		}

		public double PpmmValue
		{
			get
			{
				return ResolutionUnits.Unit.Convert(_currentUnit, ResolutionUnits.PpmmUnit, RawValue);
			}
			set
			{
				cbScaleUnits.SelectedItem = ResolutionUnits.PpmmUnit;
				RawValue = value;
			}
		}

		public double PpiValue
		{
			get
			{
				return ResolutionUnits.Unit.Convert(_currentUnit, ResolutionUnits.PpiUnit, RawValue);
			}
			set
			{
				cbScaleUnits.SelectedItem = ResolutionUnits.PpiUnit;
				RawValue = value;
			}
		}

		#endregion

		#region Private form events

		private void CbScaleUnitsSelectedIndexChanged(object sender, EventArgs e)
		{
			if (cbScaleUnits.SelectedItem != _currentUnit)
			{
				double newValue = ResolutionUnits.Unit.Convert(_currentUnit, (ResolutionUnits.Unit)cbScaleUnits.SelectedItem, RawValue);
				RawValue = newValue;
				_currentUnit = (ResolutionUnits.Unit)cbScaleUnits.SelectedItem;
			}
		}

		#endregion
	}
}
